src/core/pad.h

```diff
@@ -20,26 +20,43 @@
 #pragma once
 
 #include <stdint.h>
-
 #include <SDL.h>
+#include <GLFW/glfw3.h>
+
+#include "imgui.h" // for joypad config menu
+#include "core/system.h" // for configure() function
+#include "core/psxemulator.h"
+#include "json.hpp" // for getCfg
 
 struct PadDataS;
+using json = nlohmann::json;
 
 namespace PCSX {
 class PAD {
   public:
+
     enum pad_t { PAD1, PAD2 };
     PAD(pad_t pad);
     ~PAD();
     void init();
     void shutdown();
+    static void updateBinding(GLFWwindow* window, int key, int scancode, int action, int mods); // Actually update the binding for the button set to be configured
     unsigned char startPoll();
     unsigned char poll(unsigned char);
+    
+    json getCfg();
+    void setCfg(const json &j);
```

Comment:
There's probably a chicken-and-egg issue here when the configuration hasn't been created yet. We may want to have a "reset" method to ensure there's default values set in.